src/main/scala/io/iohk/ethereum/blockchain/sync/regular/BlockFetcher.scala

```diff
@@ -143,10 +143,15 @@ class BlockFetcher(
           log.debug("Received {} block bodies that will be ignored", bodies.size)
           state.withBodiesFetchReceived
         } else {
-          log.debug("Fetched {} block bodies", bodies.size)
-          state.withBodiesFetchReceived.addBodies(peer, bodies)
+          state.validateBodies(bodies) match {
```

Comment:
Open question: it is possible to received from the peer empty bodies at this stage? should we complain about that instead of treat it as part of the "happy" path?

Reply:
Are you asking about empty "BlockBodies" message? IIRC there's no guarantee that bodies will be fetched from the same peer and peer is free to respond only with part of data (and for sure it happens quite often with headers)

Reply:
I would maybe add comment here about this fact, that there is no guarantees about bodies being fetched from the same peer, as it can lead to scenario that the legitimate peer will send empty blockbodies response as it is on different branch.